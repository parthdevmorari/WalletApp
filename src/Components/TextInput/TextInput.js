import React from 'react';
import {TextInput} from 'react-native';
import { theme } from "../../Constants/Theme";
import styles from './TextInputStyle';
const TextField = ({
  value,
  textFieldStyle,
  multiline,
  onSubmitEditing,
  placeholder,
  onChangeText,
  keyboardType,
  editable,
  onFocus,
  onRef,
  secureTextEntry,
  error,
  onBlur,
  disable,
  autoFocus,
  length,
  ...props
}) => (
  <TextInput
    maxLength={length}
    keyboardType={keyboardType ? keyboardType : 'default'}
    secureTextEntry={secureTextEntry ? secureTextEntry : false}
    style={[
      styles.textFieldDefault,
      textFieldStyle,
      error ? {color: 'red'} : {},
    ]}
    underlineColorAndroid={'transparent'}
    onChangeText={(text) => {
      onChangeText(text);
    }}
    value={value}
	autoFocus={autoFocus}
    editable={editable}
    placeholder={placeholder}
    onSubmitEditing={() => onSubmitEditing}
    placeholderTextColor={error ? 'red' : theme.PLACEHOLDER_1}
    ref={onRef != undefined ? onRef : null}
    onFocus={() => {
      if (onFocus !== undefined) {
        onFocus();
      }
    }}
    onBlur={() => {
      if (onBlur !== undefined) {
        onBlur();
      }
    }}
    multiline={multiline}
    {...props}
  />
);

export default TextField;
