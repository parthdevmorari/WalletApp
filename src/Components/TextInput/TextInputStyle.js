import { StyleSheet, Dimensions, StatusBar } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { theme } from "../../Constants/Theme";
import { color } from '../../Constants/Color';

const width = Dimensions.get('window').width;
const ITEM_WIDTH = Math.round(width - 70);
const styles = StyleSheet.create({
	textFieldDefault: {
		height: 50,
		width: '100%',
		color:color._BLACK,
		fontSize: wp(4),
	}

});
export default styles;

