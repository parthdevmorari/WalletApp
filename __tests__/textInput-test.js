import 'react-native';
import React from 'react';
import TextInput from '../src/Components/TextInput/TextInput'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('TextArea Component', () => {
  test('TextArea Component', () => {
    const wrapper = shallow(<TextInput />);
    expect(wrapper.exists()).toBe(true);
  });
  it('TextArea  toMatchSnapshot', () => {
    const wrapper = shallow(<TextInput />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});