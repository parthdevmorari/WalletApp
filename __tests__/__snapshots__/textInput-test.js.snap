// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Component TextArea  toMatchSnapshot 1`] = `
<TextInput
  keyboardType="default"
  onBlur={[Function]}
  onChangeText={[Function]}
  onFocus={[Function]}
  onSubmitEditing={[Function]}
  placeholderTextColor="#B2B2B2"
  secureTextEntry={false}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 30,
        "height": 50,
        "width": "100%",
      },
      undefined,
      Object {},
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
